mewarpx/mewarpx/diags_store/field_diagnostic.py

```diff
@@ -38,14 +45,18 @@ def __init__(self, diag_steps, diag_data_list, grid, name, write_dir,
         self.write_dir = write_dir
         self.plot_on_diag_step = plot_on_diag_step
         self.plot_data_list = plot_data_list
+        self.post_processing = post_processing
 
         super(FieldDiagnostic, self).__init__(diag_steps, **kwargs)
 
         self.add_field_diag()
 
         if self.plot_on_diag_step:
             assert self.plot_data_list is not None, "plot_on_diag_step was True but plot_data_list is None!"
```

Comment:
See eg https://stefan.sofa-rockers.org/2018/04/16/assertions-and-exceptions/ for why to not use assert here
```suggestion
            if self.plot_data_list is None:
                raise ValueError(
                    "plot_on_diag_step was True but plot_data_list is None!")
```